unison-core/src/Unison/Var.hs

```diff
@@ -37,22 +37,25 @@ named n = typed (User n)
 refNamed :: Var v => Reference -> v
 refNamed ref = named ("ℍ" <> R.toText ref)
 
+rawName :: Type -> Text
+rawName typ = case typ of
+  User n -> n
+  Inference Ability -> "𝕖"
+  Inference Input -> "𝕒"
+  Inference Output -> "𝕣"
+  Inference Other -> "𝕩"
+  Inference PatternPureE -> "𝕞"
+  Inference PatternPureV -> "𝕧"
+  Inference PatternBindE -> "𝕞"
+  Inference PatternBindV -> "𝕧"
+  Inference TypeConstructor -> "𝕗"
+  Inference TypeConstructorArg -> "𝕦"
+  MissingResult -> "_"
+  Blank -> "_"
+  UnnamedWatch k guid -> fromString k <> "." <> guid
```

Comment:
Factored out the "create a name from a `Var.Type`" from the logic that appends the id.